Chore: More documentation (#30)
SovereignCloudStack/status-page-api: docs/overview.md

# Overview

> The `status-page-api` repository strives to provide an example implementation of the concepts being outlined by [`status-page-openapi`](https://github.com/SovereignCloudStack/status-page-openapi).

Reviewer: ```suggestion
The `status-page-api` repository strives to provide a reference implementation of the concepts being outlined by [`status-page-openapi`](https://github.com/SovereignCloudStack/status-page-openapi).
```

For the implementation [`go`](https://go.dev/) was used, as rational see the [decision record](https://github.com/SovereignCloudStack/standards/blob/main/Standards/scs-0401-v1-status-page-reference-implementation-decision.md#programming-language).

> Code under the `pkg/` directory, is considered as public example or even library code to implement your own API server, while code in `internal/` is implementation specific to this application, but can serve as example too.

Reviewer: ```suggestion
Code under the `pkg/` directory, is considered as public reference or even library code to implement your own API server, while code in `internal/` is implementation specific to this application, but can serve as reference, too.
```
